clients/drcachesim/tracer/tracer.cpp

```diff
@@ -1865,7 +1872,14 @@ event_thread_exit(void *drcontext)
                        "found active at detach.\n",
                        cur_recording_sysnum, dr_get_thread_id(drcontext));
                 // Ignore return value and try to continue in release build.
-                stop_cur_syscall_pt_trace(drcontext, data);
+                // We skip dumping the trace because the syscall was likely interrupted
+                // by the detach signal and does not represent the real app behavior.
+                // XXX: Can we somehow figure out how much of the PT trace we can
+                // keep?
+                // Such PT syscall traces at the thread's end have been seen to not
+                // decode successfully, particularly for syscalls like futex,
+                // epoll_wait.
```

Comment:
Is it possible to add a test that hits this?  Do we have unit tests for this tracing?  It would be nice to have more tests here.  Were you at least able to manually run a test that hits this path?  Please mention that in the PR description.

Reply:
> Were you at least able to manually run a test that hits this path?

Yes. PT trace from a large app had decode issues only with the last syscall's trace (the one interrupted by detach). I did multiple trace collections with this change which succeeded in decode.

> Please mention that in the PR description.

Done.

> Do we have unit tests for this tracing?

We do have unit tests for PT tracing but none of them are bursty. I added one in #7032. But there seems to be an unrelated transparency bug in the handling of futex. Originally I wanted to add the test in this PR itself, but now I decided to split it off.

> Is it possible to add a test that hits this?

Unfortunately it doesn't reproduce the decode issue I saw on the large app trace. Maybe it's because the errors I saw were mostly on internal syscalls, and only a few on futex. But still the bursty unit test is a good idea. Going to send it separately for review in #7032 as it has the weird futex transparency issue.